### DISCUSSION TITLE
Combining AND and OR with v7 selectors

### DISCUSSION THREAD

**Discussion by bbceg** [deployphp/deployer#3646] [Help needed]
So, before I used to do this:

```
task('some_task', function() {
   
})->onRoles('nginx')->onStage('stage', 'usstage')->setPrivate();
```

Initially I did this:
```
task('some_task', function() {

})->select('web_server=nginx,stage=stage,stage=usstage')->hidden();
```

Which didn't work out too well because I missed that this was an OR (doh!). I'm not sure how to achieve this in v7. As I write and compare the above I'm wondering could I do this:

```
task('some_task', function() {

})->select('web_server=nginx')->select('stage=stage,stage=usstage')->hidden();
```

This seems to work but confirmation would be appreciated (and hopefully it might be helpful for someone else).

**Comment by antonmedv** [comment]
Use here is the doc https://deployer.org/docs/7.x/selector#selector-syntax

**Comment by bbceg** [comment]
Thanks @antonmedv I have seen that doc but I couldn't see anything obvious about combining both AND and OR operators. Apologies if I've missed it.
